plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,172 @@
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) deltaLakeMetadata.getTableHandle(session, schemaTableName);
+        if (tableHandle == null) {
+            throw new TrinoException(DELTA_LAKE_INVALID_TABLE, "Table " + schemaTableName + " doesnt exist");
+        }
+        validateThatCdfWasEnabled(session, fileSystemFactory, tableHandle, sinceVersion);
+        List<DeltaLakeColumnHandle> columnHandles = deltaLakeMetadata.getColumnHandles(session, tableHandle)
+                .values().stream()
+                .map(columnHandle -> (DeltaLakeColumnHandle) columnHandle)
+                .filter(column -> !IGNORED_COLUMNS.contains(column.getName()))
+                .collect(toImmutableList());
+
+        ImmutableList.Builder<Descriptor.Field> outputFields = ImmutableList.builder();
+        columnHandles.stream()
+                .map(columnHandle -> new Descriptor.Field(columnHandle.getName(), Optional.of(columnHandle.getType())))
+                .forEach(outputFields::add);
+
+        outputFields.add(new Descriptor.Field(CHANGE_TYPE_COLUMN_NAME, Optional.of(VARCHAR)));
+        outputFields.add(new Descriptor.Field(COMMIT_VERSION_COLUMN_NAME, Optional.of(INTEGER)));
+        outputFields.add(new Descriptor.Field(COMMIT_TIMESTAMP_COLUMN_NAME, Optional.of(TIMESTAMP_TZ_MILLIS)));
+
+        return TableFunctionAnalysis.builder()
+                .handle(new TableChangesFunctionTableHandle(schemaName, tableName, sinceVersion, columnHandles))
+                .returnedType(new Descriptor(outputFields.build()))
+                .build();
+    }
+
+    public void validateThatCdfWasEnabled(
+            ConnectorSession session,
+            TrinoFileSystemFactory fileSystemFactory,
+            DeltaLakeTableHandle tableHandle,
+            long sinceVersion)
+    {
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        String transactionLogDir = getTransactionLogDir(tableHandle.getLocation());
+        long currentVersion = sinceVersion;
+
+        try {
+            while (currentVersion < tableHandle.getReadVersion()) {
+                Optional<List<DeltaLakeTransactionLogEntry>> entries = getEntriesFromJson(currentVersion, transactionLogDir, fileSystem);
+                if (entries.isPresent()) {
+                    boolean containsRemoveEntry = false;
+                    boolean containsCdcEntry = false;
+                    for (DeltaLakeTransactionLogEntry entry : entries.get()) {
+                        if (entry.getCDC() != null) {
+                            containsCdcEntry = true;
+                        }
+                        if (entry.getRemove() != null) {
+                            containsRemoveEntry = true;
+                        }
+                    }
+                    if (containsRemoveEntry && !containsCdcEntry) {
+                        throw new TrinoException(DELTA_LAKE_INVALID_TABLE, "Change Data Feed is not enabled for the requested versions range");
```

Comment:
I'd rather argue that this is a user exception (having `ErrorType.USER_ERROR`). Maybe a new `DeltaLakeErrorCode` value would be more appropriate.

Reply:
Or `StandardErrorCode.INVALID_FUNCTION_ARGUMENT` may do just fine as well.